<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>

    <arg name="robotID" default="1"/>
    <arg name="namespace" default="machine_$(arg robotID)"/>

    <arg name="host" default="localhost"/>
    <arg name="port" default="9900"/>

    <arg name="img_topic" default="video"/> <!-- camera/image_raw -->
    <arg name="camera_info_topic" default="video/camera_info"/> <!-- camera/image_raw -->
    <arg name="detections_topic" default="object_detections"/>

    <arg name="timeout_seconds" default="1"/>
    <arg name="border_dropoff" default="0.05"/>
    <arg name="feedback_topic" default="object_detections/feedback"/>

    <arg name="max_update_force" default="false"/>
    <arg name="max_update_rate" default="1.0"/>
    <arg name="network_model" default="airpose.yaml"/>
    <arg name="groundtruth" default="false"/>
    <arg name="reproject" default="true"/>

    <group ns="$(arg namespace)">
        <arg name="name" value="airpose_client"/>
        <node name="$(arg name)" pkg="airpose_client" type="airpose_client_node" output="screen"
              args="$(arg host) $(arg port)" >
            <rosparam command="load" file="$(find airpose_client)/config/$(arg network_model)"/>
            <param name="img_topic" value="$(arg img_topic)" type="string"/>
            <param name="camera_info_topic" value="$(arg camera_info_topic)" type="string"/>
            <param name="detections_topic" value="$(arg detections_topic)" type="string"/>
            <param name="feedback_topic" value="$(arg feedback_topic)" type="string"/>
            <param name="timeout_seconds" value="$(arg timeout_seconds)" type="double"/>
            <param name="border_dropoff" value="$(arg border_dropoff)" type="double"/>
            <param name="groundtruth" value="$(arg groundtruth)" type="bool"/>
            <param name="reproject" value="$(arg reproject)" type="bool"/>

            <param name="robotID" value="$(arg robotID)" type="int"/>

            <param name="max_update/force" value="$(arg max_update_force)" type="bool"/>
            <param name="max_update/rate" value="$(arg max_update_rate)" type="double"/>
        </node>
    </group>

</launch>
